packages/xr-webxr/src/feature/WebXRPlaneTracking.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Engine,
+  Matrix,
+  Quaternion,
+  Vector3,
+  XRFeatureType,
+  XRTrackingState,
+  XRPlatformPlaneTracking,
+  XRPlaneDetectionMode,
+  Logger
+} from "@galacean/engine";
+import { WebXRSessionManager } from "../WebXRSessionManager";
+import { registerXRPlatformFeature } from "../WebXRDevice";
+import { IXRTrackedPlane } from "@galacean/engine-design";
+
+@registerXRPlatformFeature(XRFeatureType.PlaneTracking)
+/**
+ *  WebXR implementation of XRPlatformPlaneTracking.
+ */
+export class WebXRPlaneTracking extends XRPlatformPlaneTracking {
+  private _sessionManager: WebXRSessionManager;
+  private _lastDetectedPlanes: XRPlaneSet;
+
+  /**
+   * Return the plane tracking mode for WebXR, which is both (Horizontal and vertical).
+   */
+  override get trackingMode(): XRPlaneDetectionMode {
+    return XRPlaneDetectionMode.EveryThing;
+  }
+
+  override set trackingMode(value: XRPlaneDetectionMode) {
+    Logger.warn("WebXR does not support modification plane detection mode.");
+  }
+
+  override _onUpdate() {
+    const { _platformFrame: platformFrame, _platformSpace: platformSpace } = this._sessionManager;
+    if (!platformFrame || !platformSpace) {
+      return;
+    }
+
+    // @ts-ignore
+    const detectedPlanes: XRPlaneSet = platformFrame.detectedPlanes || platformFrame.worldInformation?.detectedPlanes;
+    const trackedPlanes = <IWebXRTrackedPlane[]>this._trackedObjects;
+    const { _lastDetectedPlanes: lastDetectedPlanes, _added: added, _updated: updated, _removed: removed } = this;
+    added.length = updated.length = removed.length = 0;
+    for (let i = trackedPlanes.length - 1; i >= 0; i--) {
+      const trackedPlane = trackedPlanes[i];
+      const { xrPlane } = trackedPlane;
+      if (detectedPlanes?.has(xrPlane)) {
+        if (trackedPlane.lastChangedTime < xrPlane.lastChangedTime) {
+          this._updatePlane(platformFrame, platformSpace, trackedPlane, xrPlane);
+          updated.push(trackedPlane);
+        }
+      } else {
+        trackedPlanes.splice(i, 1);
+        trackedPlane.state = XRTrackingState.NotTracking;
+        trackedPlane.lastChangedTime = 0;
+        trackedPlane.xrPlane = null;
+        removed.push(trackedPlane);
+      }
+    }
+    detectedPlanes.forEach((xrPlane) => {
```

Comment:
Do not use `forEach` in `onUpdate` Function

Reply:
![image](https://github.com/galacean/runtime/assets/7768919/b7586bf6-ceb9-4029-9ef5-d91b74b1ecdf)
This is not an array, it is a set.